Update the README

aevitas opened the pull request:
Most notable departure is the positioning of Fern as a toolset that doesn't compete with OpenAPI, but works alongside it and provides strong code generation features.

dannysheridan commented on README.md:
```diff
@@ -16,72 +16,72 @@ Backed by Y Combinator
+**Fern** is an **open source toolset** for **defining, building and generating APIs and SDKs**. Fern can be used in your existing projects through OpenAPI, and integrate effortlessly into your existing workflows. It can also serve as the basis of your API development, by describing your endpoints in a Fern specification, and having Fern generate your API backend code for you.
 
-### 🌿 SDKs
+This repository contains the Fern **compiler**. The compiler transforms an API definition, either in Fern or OpenAPI format into various code artifacts, such as Client Libraries, SDKs, or even fully-fledged **Express** or **FastAPI** backends.
```
Express, FastAPI, or Spring Boot

dannysheridan replied:
lowercase because not a proper noun

dannysheridan replied:
full-fledged

dannysheridan commented on README.md:
If you describe what you think should be in the header image, `header.png`, I'm happy to work with a graphic designer to make it.

aevitas replied:
I think simple would do the trick here - just the project's logo and name, along with a nice tagline that describes the project in one go. I'll see if I can think of a few.

dannysheridan replied:
The most concise definition to me: Define your API. Generate API documentation, clients, and server code.

dannysheridan replied:
![logo_no background_black text](https://github.com/fern-api/fern/assets/83524670/1f9455ce-1c42-44d9-9bc3-fbde0b560bcb)
![logo_no background_white text](https://github.com/fern-api/fern/assets/83524670/37fbc0e5-1e6f-4b22-9bf0-e2216bc38f50)

dannysheridan commented on README.md:
```diff
@@ -16,72 +16,72 @@ Backed by Y Combinator
+# Fern
 
-This repository contains the **Fern compiler.** The compiler transforms the API description into useful outputs, like:
+**Fern** is an **open source toolset** for **defining, building and generating APIs and SDKs**. Fern can be used in your existing projects through OpenAPI, and integrate effortlessly into your existing workflows. It can also serve as the basis of your API development, by describing your endpoints in a Fern specification, and having Fern generate your API backend code for you.
```
let's call it an "open source toolkit" instead of toolset (more common term)

dannysheridan replied:
Alternative phrasing suggestion:Fern is an open source toolkit for building API documentation and SDKs. Bring your existing OpenAPI definition or describe your endpoints in the Fern specification. Fern automatically generates API documentation, client libraries, and backend code.

aevitas replied:
> let's call it an "open source toolkit" instead of toolset (more common term)

Agreed. I do want to emphasize Fern's ability to generate backend code as well, what do you think of this updated version?

dannysheridan commented on README.md:
```diff
@@ -16,72 +16,72 @@ Backed by Y Combinator
+**Fern** is an **open source toolset** for **defining, building and generating APIs and SDKs**. Fern can be used in your existing projects through OpenAPI, and integrate effortlessly into your existing workflows. It can also serve as the basis of your API development, by describing your endpoints in a Fern specification, and having Fern generate your API backend code for you.
 
-### 🌿 SDKs
+This repository contains the Fern **compiler**. The compiler transforms an API definition, either in Fern or OpenAPI format into various code artifacts, such as Client Libraries, SDKs, or even fully-fledged **Express** or **FastAPI** backends.
```
should we use `specification` to be consistent?

aevitas replied:
Call API definitions specifications everywhere you mean?

dsinghvi replied:
I like definition better than specification.

dsinghvi replied:
The reasoning why is b/c API Definition refers to the actual API (yaml files in Fern, or openapi.yml) and specification refers to the schema of the definition (i.e. what things can the definition be about).

aevitas replied:
Therefore: the Fern schema is a specification distinct from the OpenAPI specification, but a specific API defined using either specification would be a definition of said API?

dsinghvi replied:
thats right ^

dannysheridan commented on README.md:
```diff
@@ -16,72 +16,72 @@ Backed by Y Combinator
+**Fern** is an **open source toolkit** for defining, building and generating **APIs, documentation and SDKs**. Bring an existing OpenAPI definition or describe your endpoints in a Fern specification. Fern generates backend code, specifications in various formats, and client libraries with a single command.
 
-### 🌿 SDKs
+This repository contains the Fern **compiler**. The compiler transforms an API definition into various code artifacts, such as client libraries, SDKs, or even full-fledged **Express**, **Fast API** or **Spring Boot** backends.
```
FastAPI

dsinghvi replied:
> This repository contains the Fern compiler. The compiler transforms an API definition, either in Fern or OpenAPI format into various code artifacts, such as Client Libraries, SDKs, or even fully-fledged Express or FastAPI backends.

I like the original paragraph here better, i think its important to mention both the `OpenAPI and Fern format`. Since it's jus the intro, i'm okay with just listing 2 backend frameworks as examples.

aevitas replied:
I've mixed and matched the two - how do you like it now?

dannysheridan commented on README.md:
```diff
@@ -16,72 +16,72 @@ Backed by Y Combinator
+# Fern
 
-This repository contains the **Fern compiler.** The compiler transforms the API description into useful outputs, like:
+**Fern** is an **open source toolkit** for defining, building and generating **APIs, documentation and SDKs**. Bring an existing OpenAPI definition or describe your endpoints in a Fern specification. Fern generates backend code, specifications in various formats, and client libraries with a single command.
```
this leaves me thinking: which ones?

dsinghvi replied:
I liked the original paragraphs better than what we have now: 

> Fern is an open source toolset for defining, building and generating APIs and SDKs. Fern can be used in your existing projects through OpenAPI, and integrate effortlessly into your existing workflows. It can also serve as the basis of your API development, by describing your endpoints in a Fern specification, and having Fern generate your API backend code for you.

> This repository contains the Fern compiler. The compiler transforms an API definition, either in Fern or OpenAPI format into various code artifacts, such as Client Libraries, SDKs, or even fully-fledged Express or FastAPI backends.

> In short, Fern takes care of the boilerplate associated with developing APIs, so you can focus on delivering business value.

dsinghvi replied:
What do you think about starting with a sentence like this: 
> Fern is an open source toolset for designing, building and consuming REST APIs. 

Here's how i think about the 3 phases: 

**Design** Developers can opt to design their APIs in the Fern Specification. If they'd rather use OpenAPI though, we support reading that as well. 

**Build** Fern generates boilerplate code for backends written in FastAPI, Express, and Spring Boot. Fern generates all the networking logic and provides type safety, so that developers can focus on the business logic. 

**Consume** Fern generates client libraries for multiple programming languages. We've built the client library generators from the ground up so that they follow the idioms of each language and feel handwritten.

dannysheridan commented on README.md:
```diff
@@ -16,72 +16,72 @@ Backed by Y Combinator
+# Fern
 
-This repository contains the **Fern compiler.** The compiler transforms the API description into useful outputs, like:
+**Fern** is an **open source toolkit** for defining, building and generating **APIs, documentation and SDKs**. Bring an existing OpenAPI definition or describe your endpoints in a Fern specification. Fern generates backend code, specifications in various formats, and client libraries with a single command.
```
not sure this is needed

dsinghvi replied:
what text is this referring to?

dannysheridan commented on README.md:
```diff
@@ -16,72 +16,72 @@ Backed by Y Combinator
+### 🌿 Backend Code Generation
 
-You can feed the generated OpenAPI into the endless list of tools that support OpenAPI.
+Define your API, and Fern will generate the boilerplate for you. All the generated code is validated at build time, ensuring your all your endpoints and their data are being served correctly.
```
would be specific, such as mentioning Fern will generate the models and networking logic. Just add your business logic (also called application logic by devs, not sure which term is better)

aevitas replied:
I think business logic is the broader term, at least in DDD and such. I reckon "endpoints" is more applicable than networking logic here, because REST APIs rely on HTTP for their networking, we're just implementing an endpoint.

dannysheridan commented on README.md:
```diff
@@ -16,72 +16,72 @@ Backed by Y Combinator
+In short, Fern takes care of the boilerplate associated with developing APIs, so you can **focus on delivering business value**.
 
-### 🌿 Server-side code generation
+# Capabilities
```
Versioning an API, SDKs, and docs comes up often in customer calls. Where do you think we ought to mention 1-2 sentences about versioning?

dsinghvi replied:
I think versioning isn't an important enough topic to include the README, i can see us leaving that in the docs somewhere. (and fern doesn't _really_ solve this problem since its pretty unopinionated about versioning)

dannysheridan commented on README.md:
```diff
@@ -16,72 +16,72 @@ Backed by Y Combinator
-- [Focus on server-side API development](https://www.buildwithfern.com/docs/comparison-with-openapi#focus-on-server-side-api-development)
-- [Change management](https://www.buildwithfern.com/docs/comparison-with-openapi#change-management)
-- [Cloud-based code generation and publishing](https://www.buildwithfern.com/docs/comparison-with-openapi#cloud-based-code-generation-and-publishing)
+We have designed Fern to complement existing OpenAPI toolchains and workflows, rather than replace them. We believe OpenAPI does a good job at as a declarative standard for defining APIs, but at times it is too verbose and complex, which harms the quality of automatically generated code.
```
love this! kudos on clear writing

dannysheridan commented on README.md:
```diff
@@ -16,72 +16,72 @@ Backed by Y Combinator
+    
+-   [﻿Cloud-based code generation and publishing](https://www.buildwithfern.com/docs/comparison-with-openapi#cloud-based-code-generation-and-publishing)
+    
+# Getting Started
```
going back and forth between "if you have an openapi" and "if you are starting from scratch" is hard for me to process as a reader.

think it makes sense to have two getting started sections here?

aevitas replied:
Agreed, this was something I was struggling with when writing. Moving `--openapi`  into a separate OpenAPI section.

dannysheridan commented on README.md:
```diff
@@ -129,102 +121,126 @@ service:
 
-You can add [generators](compiler/generators) using `fern add`. By default, this
-will publish your SDK to the Fern npm registry (npm.buildwithfern.com).
+You'll be prompted to log in in a new browser tab using your GitHub account.
```
*in to

dannysheridan commented on README.md:
```diff
@@ -129,102 +121,126 @@ service:
+  local:
+    generators:
+      - name: fernapi/fern-typescript-node-sdk
+        version: 0.6.4
```
want to use the latest release? 0.7.1

dannysheridan commented on README.md:
````diff
@@ -129,102 +121,126 @@ service:
+```
 
-### `fern generate [--group <group>] [--version <version>]`
+Note the `local-file-system` output may not be present by default. If it's not, simply add it and specify a local path to where the generated client will be saved.
````
Do you think we should add this in by default? Can make the change on the `fern add fern-typescript-sdk` command if so.

dannysheridan replied:
I would think that we should show the default behavior and minimize concepts taught on the first pass.

dsinghvi replied:
Let's just treat CLI changes as out of scope for this PR (going to resolve this)

dannysheridan commented on README.md:
```diff
@@ -129,102 +121,126 @@ service:
 
-If you don't have a `default-group` specified in `generators.yml`, then you must
-specify a group using the `--group` option.
+Fern supports automatically publishing clients to an NPM registry. By default, Node packages are published to `npm.buildwithfern.com` and are named `@organisation-fern/package`
```
Want to include this line from the website?

`Whether you call them SDKs, client libraries, bindings, or wrappers, Fern generates prewritten code that developers can use to‍ build client applications more easily.`

aevitas replied:
I've included it in the capabilities section instead, felt like the right place for it.

dannysheridan commented on README.md:
```diff
@@ -129,102 +121,126 @@ service:
 
-If you don't have a `default-group` specified in `generators.yml`, then you must
-specify a group using the `--group` option.
+Fern supports automatically publishing clients to an NPM registry. By default, Node packages are published to `npm.buildwithfern.com` and are named `@organisation-fern/package`
```
`@organization-fern/package` -> US English spelling

aevitas replied:
🇺🇸🇺🇸🇺🇸

dannysheridan commented on README.md:
```diff
@@ -129,102 +121,126 @@ service:
-APIs. Read more in our [docs](https://www.buildwithfern.com/docs/advanced/depending-on-other-apis).
+| Command                | Description                                                                                                                                                                                                     |
+|------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `fern init`            | Initialised a new Fern workspace.                                                                                                                                                                               |
```
Initialize -> US English spelling

dannysheridan commented on README.md:
```diff
@@ -16,72 +16,72 @@ Backed by Y Combinator
-- [Focus on server-side API development](https://www.buildwithfern.com/docs/comparison-with-openapi#focus-on-server-side-api-development)
-- [Change management](https://www.buildwithfern.com/docs/comparison-with-openapi#change-management)
-- [Cloud-based code generation and publishing](https://www.buildwithfern.com/docs/comparison-with-openapi#cloud-based-code-generation-and-publishing)
+We have designed Fern to complement existing OpenAPI toolchains and workflows, rather than replace them. We believe OpenAPI does a good job at as a declarative standard for defining APIs, but at times it is too verbose and complex, which harms the quality of automatically generated code.
```
We have designed Fern to complement existing OpenAPI tools and workflows rather than replace them. We believe that OpenAPI has helped the development community by providing a standard for expressing REST APIs. Still, at times it is verbose and has [footguns](https://en.wiktionary.org/wiki/footgun) that, if used, make it impossible to generate idiomatic code.

dsinghvi replied:
> make it impossible to generate idiomatic code 

This is probably too exaggerated

dannysheridan commented on README.md:
```diff
@@ -16,72 +16,72 @@ Backed by Y Combinator
+Currently, the Fern compiler can generate the following types of artifacts:
 
-### 🌿 Postman Collection
+### 🌿 Client Libraries and SDKs
```
Want to mention the languages we support? (TS, Python, Java, Go) and suggest that folks +1 a GitHub Issue for ones that are on the roadmap (C#, Ruby, Rust)

dsinghvi replied:
I think its also worth explicitly acknowledging that we are using two terms here to mean the same thing "Client LIbrary" and "SDK".

dannysheridan commented on README.md:
```diff
@@ -16,72 +16,72 @@ Backed by Y Combinator
+Fern generates idiomatic SDKs you can use to interact with your APIs, or publish for third party usage. This makes keeping your client libraries up to date with your API endpoints trivial, and speeds up your workflows dramatically.
 
-### 🌿 An OpenAPI spec
+### 🌿 Backend Code Generation
```
Want to mention the ones we support? 

Express, FastAPI, Spring Boot (don't think we need to mention DropWizard or Jersey)

dannysheridan commented on README.md:
```diff
@@ -16,72 +16,72 @@ Backed by Y Combinator
+### 🌿 Developer Resources
 
-OpenAPI is a great tool for documenting APIs, but falls short for code generation use cases.
+Fern supports the generation of development resources such as Postman collections, and OpenAPI specifications, making integration into existing development workflows effortless.
```
Fern supports generating an OpenAPI specification and Postman collection, making it easy to keep your docs and Postman workspace up to date.

dsinghvi replied:
For the purpose of this README, I think worth focusing on OpenAPI as an "input"

dsinghvi commented on README.md:
```diff
@@ -16,72 +16,72 @@ Backed by Y Combinator
+Define your API, and Fern will generate the boilerplate for you. All the generated code is validated at build time, ensuring your all your endpoints and their data are being served correctly.
 
-# Comparison with OpenAPI
+### 🌿 Developer Resources
```
Developer Resources is pretty vague -- what do you think of using the word `documentation`  instead? 

The two things here that I would want to highlight are 
1/ Fern supports generating Postman collections with examples of successful and unsuccessful requests/respones. 
2/ Fern supports generating Documentation websites (see here: https://docs.vellum.ai/api-reference/registered-prompts/register-prompt)

dsinghvi commented on README.md:
```diff
@@ -129,102 +121,126 @@ service:
+-   [﻿Generated TypeScript SDK](https://github.com/fern-api/plantstore-node)    
+-   [﻿Generated Java SDK](https://github.com/fern-api/plantstore-java)    
+-   [﻿Generated Postman Collection](https://github.com/fern-api/plantstore-postman)    
+-   [﻿Generated OpenAPI spec](https://github.com/fern-api/plantstore-openapi)
```
@aevitas i could see ditching the Plantstore example and showing real examples of folks using Fern. 

Dropping some links for you here: 
- Flatfile TypeScript SDK (https://github.com/FlatFilers/flatfile-node)
- Vellum Python SDK (https://pypi.org/project/vellum-ai/) 
- Vellum Docs (https://docs.vellum.ai/api-reference/generate)
- Flipt Java SDK (https://github.com/flipt-io/flipt-java)

aevitas replied:
For sure - I've replaced the plant store with the real world examples.

dsinghvi commented on README.md:
```diff
@@ -14,74 +17,78 @@
+# Fern
+
+Fern is an open source toolkit for designing, building and consuming REST APIs. Fern can be used in your existing projects through OpenAPI, and integrate effortlessly into your existing workflows. It can also serve as the basis of your API development, by describing your endpoints in a Fern specification, and having Fern generate your API backend code, as well as SDKs.
+
```
I think we should get to the punchline a little earlier: 

Fern is an open source toolkit for designing, building and consuming REST APIs. With Fern, you can generate documentation, client libraries and boilerplate for your backend server in your chosen framework (e.g. FastAPI, Express). 

Fern can be used in your existing projects through OpenAPI or if you choose, you can describe your API in the Fern specification. In this sense, Fern is an alternative to the OpenAPI standard without seeking to replace it.

This repository contains the Fern compiler. The compiler takes...

dsinghvi replied:
@aevitas what do you think about this for the intro?

dsinghvi commented on README.md:
```diff
@@ -93,34 +100,24 @@ fern/
-        type: double
-        docs: The rating scale is one to five stars
-
+      title: string rating:
```
rating should be on a new line

dsinghvi commented on README.md:
```diff
@@ -93,34 +100,24 @@ fern/
-        docs: The rating scale is one to five stars
-
+      title: string rating:
+        type: double docs: The rating scale is one to five stars
```
docs should be on a newline

dsinghvi commented on README.md:
```diff
@@ -93,34 +100,24 @@ fern/
-service:
-  auth: false
-  base-path: /movies
+      title: string rating: double service:
```
fix newlines

dsinghvi commented on README.md:
```diff
@@ -129,102 +126,134 @@ service:
+## Publishing Clients
 
-### `fern check`
+Fern supports automatically publishing clients to an NPM registry. By default, Node packages are published to `npm.buildwithfern.com` and are named `@organization-fern/package`
```
can we embed an example config of `generators.yml` here? 

https://github.com/fern-api/fern-definition-registry/blob/4aabec28789ba7de517159053f2568b95803b46c/fern/api/generators.yml#L21-L26

aevitas replied:
Done. Should I mention that different publishing targets are part of the paid version of Fern?

dsinghvi commented on README.md:
```diff
@@ -14,74 +17,78 @@
+# Getting Started
 
-# Get started
+The Fern tools are available as a Node package. To install it, simply run:
```
if we are going to have a CLI Reference at the bottom of the readme then i think we should be consistent with terminology -- so let's use CLI instead of `Node package`

aevitas replied:
Agreed.

dsinghvi reviewed (approved):
@aevitas added a couple more comments, once we resolve those, im down to merge this and make changes on top

aevitas commented on README.md:
```diff
@@ -93,34 +100,24 @@ fern/
-  auth: false
-  base-path: /movies
+      title: string rating: double service:
+  auth: false  base-path: /movies
```
Fixed them all - it was an editor I used briefly that messed this up.

zachkirsch commented on README.md:
```diff
@@ -14,74 +18,78 @@
----
+# Fern
+
+Fern is an open source toolkit for designing, building and consuming REST APIs. With Fern, you can generate client libraries, documentation and boilerplate for your backend server in your chosen framework (e.g. FastAPI, Express).
```
How about

> With Fern, you can generate client libraries, documentation and boilerplate for your backend server.

aevitas replied:
To be fair, I think that would be a great description for the repository. In the readme though, I do want to emphasize Fern supports you through all stages of developing and maintaining an API a little bit more. Definitely the most succinct way to put it though!

dsinghvi replied:
> With Fern, you can generate client libraries, documentation and boilerplate for your backend server.

I could see us mentioning REST APIs. 

Fern is an open source toolkit for REST APIs. With Fern, you can generate client libraries, documentation and boilerplate code for your backend server.

aevitas commented on README.md:
```diff
@@ -129,102 +126,134 @@ service:
 
-You can add [generators](compiler/generators) using `fern add`. By default, this
-will publish your SDK to the Fern npm registry (npm.buildwithfern.com).
+The path to the definition can either be a local file, or a HTTP resource. If you start from an OpenAPI definition, the `init` command will forego generating the default project files such as `imbd.yml`, and instead will use your definition as a starting point.
```
Done - though this currently produces an error. I have opened issue #1845 to track this.

dsinghvi commented on README.md:
```diff
@@ -14,74 +18,78 @@
+- TypeScript
+- Java
+- Python
+- Go
```
lets link to each of the repos here? (`fern-api/fern-typescript` and so on?)

dsinghvi commented on README.md:
```diff
@@ -14,74 +18,78 @@
+Define your API, and Fern will generate models, endpoints and boilerplate application code. All the generated code is validated at build time, ensuring your all your endpoints and their data are being served correctly.
 
-You can feed the generated OpenAPI into the endless list of tools that support OpenAPI.
+We currently supported:
```
Fern currently supports*

aevitas replied:
YIkes.

aevitas commented:
@zachkirsch What do you think of this header?

dsinghvi commented on README.md:
```diff
@@ -1,87 +1,90 @@
+### 🌿 Client Libraries and SDKs
 
-Client libraries speed up internal development, and help acquire customers who use your API. Our auto-generated SDKs are idiomatic and feel handwritten.
+Whether you call them SDKs, client libraries, bindings, or wrappers, Fern generates idiomatic SDKs you can use to interact with your APIs, or publish for third party usage. This makes keeping your client libraries up to date with your API endpoints trivial, and speeds up your workflows dramatically.
```
I think in this para we want to: 
- clarify that what we call SDK could also be referred to as client library, binding, wrapper
- mention that we've focused on the quality of the generated code (as if a human wrote it)
- generated SDKs contain types, networking logic, error handling so that you can *consume* the API effortlessly

dsinghvi commented on README.md:
```diff
@@ -1,87 +1,90 @@
-# Comparison with OpenAPI
+- Express
+- Spring Boot
+- FastAPI
```
lets specify the languages next to each and if they want another server framework we can point to an issue as well 

Express (TypeScript)
FastAPI (Python)
Spring Boot (Java)

dsinghvi commented on README.md:
```diff
@@ -1,87 +1,90 @@
+### 🌿 Backend Code
 
-### 🌿 An OpenAPI spec
+Define your API, and Fern will generate models, endpoints and boilerplate application code. All the generated code is validated at build time, ensuring your all your endpoints and their data are being served correctly.
```
> Define your API, and Fern will generate models, endpoints and boilerplate application code.

Fern will generate the models, but only part of the endpoints. An endpoint is made up of networking logic + business logic and fern handles all the boilerplate networking logic like parsing path parameters, query parameters, sending errors.

dsinghvi replied:
> All the generated code is validated at build time, ensuring your all your endpoints and their data are being served correctly.

Is "compile" time better than "build" time? Also the generated code is not "validated" but the endpoint implementation is validated. 

Maybe something like: 

The generated server code provides autocompletion and type-safety, ensuring that all your endpoints are served correctly and not diverging from the API definition. For example, if you return the wrong type for an endpoint, you’ll get a compile error.

dsinghvi commented on README.md:
````diff
@@ -136,95 +139,146 @@ errors:
+For example, you can use the following command to start a new Fern workspace based on an OpenAPI Pet Store definition:
+
+```
+fern init --openapi fern init --openapi https://petstore.swagger.io/v2/swagger.json
````
incorrect command: 

```bash
fern init --openapi https://petstore.swagger.io/v2/swagger.json
```

dsinghvi commented on README.md:
```diff
@@ -136,95 +139,146 @@ errors:
+## Adding Generators
 
-### `fern generate [--group <group>] [--version <version>]`
+You can add generators using the `fern add`. You can have multiple generators for a single project, e.g. a single `fern generate` call could generate SDKs in Python, TypeScript and Java all in one go.
```
using the command `fern add <generator>`

dsinghvi commented on README.md:
```diff
@@ -136,95 +139,146 @@ errors:
 
-You can specify a version using the `--version` option. This version string is
-used when publishing SDKs to registries (e.g. npm).
+Which will add the Node variant of the TypeScript SDK to the list of generators - the other variant being a TypeScript browser library, to be used by frontends directly.
```
This will*

dsinghvi commented on README.md:
````diff
@@ -136,95 +139,146 @@ errors:
-If you don't have a `default-group` specified in `generators.yml`, then you must
-specify a group using the `--group` option.
+```
+fern add fern-typescript-sdk
````
lets do `fern add fern-typescript-node-sdk`, i wonder if this also fixes the petstore issue you were running into

aevitas replied:
Fixed - I think it was already using that generator though when it failed: 

```
default-group: local
groups:
  local:
    generators:
      - name: fernapi/fern-typescript-node-sdk
        version: 0.7.1
        output:
          location: local-file-system
          path: ../../generated/typescript
      - name: fernapi/fern-openapi
        version: 0.0.28
        output:
          location: local-file-system
          path: ../../generated/openapi

```

aevitas commented:
> @aevitas can you run `yarn format:fix` to update any formatting issues

I would, but this literally leads to hundreds of files being modified? 

Files like:

```
        modified:   packages/ui/deploy/src/createCdkStack.ts
        modified:   packages/ui/deploy/src/environment.ts
        modified:   packages/ui/deploy/src/index.ts
        modified:   packages/ui/deploy/src/stack.ts
        modified:   packages/ui/deploy/tsconfig.json
        modified:   packages/ui/fe-bundle/.depcheckrc.json
        modified:   packages/ui/fe-bundle/.env-cmdrc.cjs
        modified:   packages/ui/fe-bundle/.mrlint.json
```

dsinghvi commented:
> > @aevitas can you run `yarn format:fix` to update any formatting issues
> 
> I would, but this literally leads to hundreds of files being modified?
> 
> Files like:
> 
> ```
>         modified:   packages/ui/deploy/src/createCdkStack.ts
>         modified:   packages/ui/deploy/src/environment.ts
>         modified:   packages/ui/deploy/src/index.ts
>         modified:   packages/ui/deploy/src/stack.ts
>         modified:   packages/ui/deploy/tsconfig.json
>         modified:   packages/ui/fe-bundle/.depcheckrc.json
>         modified:   packages/ui/fe-bundle/.env-cmdrc.cjs
>         modified:   packages/ui/fe-bundle/.mrlint.json
> ```

Ah interesting

zachkirsch commented:
@aevitas I made some edits but had trouble making a PR to your fork. What do you think? [README.md](https://github.com/fern-api/fern/files/11972860/README.md)

aevitas commented:
> @aevitas I made some edits but had trouble making a PR to your fork. What do you think? [README.md](https://github.com/fern-api/fern/files/11972898/README.md)

LGTM! I've merged almost all of them, @dsinghvi can you take one final look?

dsinghvi commented:
> > @aevitas I made some edits but had trouble making a PR to your fork. What do you think? [README.md](https://github.com/fern-api/fern/files/11972898/README.md)
> 
> LGTM! I've merged almost all of them, @dsinghvi can you take one final look?

@aevitas are you cool with just using @zachkirsch's copy to start? and then we can make separate follow up PRs if there are things that are missing. this PR has gotten large enough so hard to see what you've changed from his version
